app/react/Templates/components/specs/PropertyOption.spec.js

```diff
@@ -32,6 +32,24 @@ describe('PropertyOption', () => {
       const option = TestUtils.findRenderedComponentWithType(component, PropertyOption).getWrappedInstance();
       expect(option.props.removeProperty).toEqual(jasmine.any(Function));
     });
+
+    it('should not add a property when clicked if disabled', () => {
+      const props = { label: 'test', disabled: true, type: 'optionType', addProperty: jasmine.createSpy().and.returnValue({}) };
+      TestComponent = wrapInTestContext(dragSourceOption);
+      component = renderComponent(TestComponent, props);
+      const button = TestUtils.findRenderedDOMComponentWithTag(component, 'button');
+      TestUtils.Simulate.click(button);
+      expect(props.addProperty).not.toHaveBeenCalled();
+    });
+
+    it('should add a property when clicked', () => {
+      const props = { label: 'test', type: 'optionType', addProperty: jasmine.createSpy().and.returnValue({}) };
+      TestComponent = wrapInTestContext(dragSourceOption);
+      component = renderComponent(TestComponent, props);
+      const button = TestUtils.findRenderedDOMComponentWithTag(component, 'button');
+      TestUtils.Simulate.click(button);
+      expect(props.addProperty).toHaveBeenCalled();
```

Comment:
While line 42: not.toHaveBeenCalled may be enough, just 'toHaveBeenCalled' when there are specific parameters passed to the function call is not comprehensive enough.  This test would be green if I removed the argument for the call.  

Please ensure that this is a toHaveBeenCalledWith that tests the relevant arguments are passed to the function.